assets/questions/bomberman.js

```diff
@@ -0,0 +1,148 @@
+                    if enemiesKilled[r][c] > maxEnemiesKilled:
+                        maxEnemiesKilled = enemiesKilled[r][c]
+        return maxEnemiesKilled
+
+    @classmethod
+    def bombAllEnemiesOnBoard(cls, board):
+        if not board or not board[0]:
+            return 0
+        count = 0
+        for r in range(len(board)):
+            for c in range(len(board[0])):
+                if board[r][c] == "e":
+                    count += 1
+        return count
+`
+  },
+  tasks: [{
+    instructions: [
+      [
+        'For this question, you will be given a board, in which all elements are strings. ',
+        '"x" represents a wall, "e" represents an enemy, and "" represents an empty cell. ',
+        'You can place an bomb in an empty cell, which will kill all enemies in vertical and horizontal ',
+        'directions until the blast hits a wall or the edge of the board.',
+        'Write a function bomb(board), to find the best position to place a bomb in order to',
+        'kill as many enemies as possible, then return the number of enemies killed.'
+      ].join('')
+    ],
+    prerequisiteSkills: ['Arrays', 'Strings'],
+    acquiredSkills: ['Dynamic Programming'],
+    inputFunctionName: null,
+    outputFunctionName: null,
+    mainFunctionName: 'bomb',
+    correctnessTests: [
+      {
+        input: [],
+        allowedOutputs: [0]
+      },
+      {
+        input: [[], [], []],
+        allowedOutputs: [0]
+      },   
+      {
+        input: [["e", "", "e", "x", "", "e"]],
+        allowedOutputs: [2]
+      },
+      {
+        input: [["e", "", "e", "", "", "e"]],
+        allowedOutputs: [3]
+      },
+      {
+        input: [["e"], ["x"], [""], ["x"]],
+        allowedOutputs: [0]
+      },
+      {
+        input: [["e"], ["e"], [""], ["x"]],
+        allowedOutputs: [2]
+      },
+      {
+        input: [["", "e", ""], ["e", "", "e"], ["", "e", ""]],
+        allowedOutputs: [4]
+      }
+    ],
+    buggyOutputTests: [{
+      buggyFunctionName: 'AuxiliaryCode.ignoreWalls',
+      messages: [["Try to run your code on [['e', '', 'x', 'e']]. ",
+        "Does the result match what you expected? "
+      ].join(''),
+        [
+          "You seem to be ignoring walls in your solution, which is incorrect.",
+          "If a bomb is placed, the resulting explosion should only go until it reaches a wall."
+        ].join('')
+      ]
+    },
+    {
+      buggyFunctionName: 'AuxiliaryCode.bombAllEnemiesOnBoard',
+      messages: [["Try to run your code on [['e', 'x', '', 'x', 'e']]. ",
+        "Does the result match what you expected?"
+      ].join(''),
+        [
+          "You seemed to be returning all the enemies on the board. ",
+          "Your bomb can only kill enemies in the same row and column. "
+        ].join('')
+      ]
+    }],
+    // Since we currently only support O(n) runtime complexities and this
+    // requires an O(n^2) solution. We have skipped performance tests.
```

Comment:
This is a good place to file a TODO, but you should also file an issue for adding performance tests for the Bomberman problem.

Also, The system currently only supports O(n) runtime complexities and this requires an O(n^2) solution, so performance tests will be implemented at a later date.